ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -241,7 +241,14 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u", (t.neg ? "-" : ""), t.hour, t.minute, t.second);
+    if (field->decimals > 0) {
+        char * format;
+        mnd_sprintf(&format, 0, "%%s%%02u:%%02u:%%02u.%%%02uu", field->decimals);
```

Comment:
Instead of creating a dynamic format string, you should be able to use something like `%0*u` to specify the width dynamically.

Reply:
Thanks.
Guess I could use some more libc knowledge. :)
Will do.